logging/google/cloud/logging/handlers/transports/background_thread.py

```diff
@@ -254,9 +253,7 @@ class BackgroundThreadTransport(Transport):
 
     def __init__(self, client, name, grace_period=_DEFAULT_GRACE_PERIOD,
                  batch_size=_DEFAULT_MAX_BATCH_SIZE):
-        http = copy.deepcopy(client._http)
-        self.client = client.__class__(
-            client.project, client._credentials, http)
+        self.client = client
```

Comment:
Is the copy completely un-needed now? Was the lack of thread-safety in `httplib2` the sole reason for copying?

Reply:
Yeah that was my wild guess here and it seems it was correct. :)

Reply:
I didn't have much context about the reason for deepcopying the http object. I have tried reproduce this using `google-cloud-core==0.26.0`(which switched from `httplib2` to `requests` as the HTTP transport) and `google-cloud-logging==1.2.0`(which depends on core 0.26.0), and the same exception occurred. But it would not occur with an earlier version. And removing the deepcopy of http can solve the issue here. Still investigating if there are any other reason for deepcopying and ensure it's safe to remove it. Maybe the original author @waprin can share some idea here?

Reply:
We indeed deepcopied this because of thread-safety concerns.

Reply:
what @jonparrott said :)